libvast/vast/hasher.hpp

```diff
@@ -0,0 +1,193 @@
+  : public hasher<simple_hasher<HashFunction>,
+                  typename HashFunction::result_type>,
+    vast::detail::equality_comparable<simple_hasher<HashFunction>> {
+  using super
+    = hasher<simple_hasher<HashFunction>, typename HashFunction::result_type>;
+
+public:
+  /// Constructs a hasher from a vector of hash functions.
+  /// @param k The number of hash functions to use.
+  /// @param xs The vector with seeds for the hash functions.
+  /// @pre `!xs.empty()`
+  explicit simple_hasher(size_t k = 1, std::vector<size_t> xs = {0})
+    : super{k}, seeds_{std::move(xs)} {
+    VAST_ASSERT(k == seeds_.size());
+  }
+
+  /// Hashes a value *k* times with *k* hash functions.
+  /// @param T x The value to hash.
+  /// @param Ts xs The sequence to write the digests into.
+  template <class T, class Ts>
+  void hash(const T& x, Ts& xs) {
+    VAST_ASSERT(xs.size() == seeds_.size());
+    for (size_t i = 0; i < xs.size(); ++i)
+      xs[i] = detail::seeded_hash<HashFunction>{seeds_[i]}(x);
+  }
+
+  // -- concepts -------------------------------------------------------------
+
+  friend bool operator==(const simple_hasher& x, const simple_hasher& y) {
+    return x.seeds_ == y.seeds_;
+  }
+
+  template <class Inspector>
+  friend auto inspect(Inspector& f, simple_hasher& x) {
+    return f(static_cast<super&>(x), x.seeds_);
+  }
+
+private:
+  std::vector<size_t> seeds_;
+};
+
+/// A hasher that uses *double hashing* to compute multiple digests.
+template <class HashFunction>
+class double_hasher
+  : public hasher<double_hasher<HashFunction>,
+                  typename HashFunction::result_type>,
+    vast::detail::equality_comparable<double_hasher<HashFunction>> {
+  using super
+    = hasher<double_hasher<HashFunction>, typename HashFunction::result_type>;
+
+public:
+  /// Constructs a hasher with a vector of seeds.
+  /// @param k The number of hash functions to use.
+  /// @param xs The vector with seeds for the hash functions.
+  explicit double_hasher(size_t k = 2, const std::vector<size_t>& xs = {0, 1})
+    : super{k} {
+    VAST_ASSERT(xs.size() == 2);
+    seed1_ = xs[0];
+    seed2_ = xs[1];
+  }
+
+  /// Hashes a value *k* times with *2* hash functions via *double hashing*.
+  /// @param T x The value to hash.
+  /// @param Ts xs The sequence to write the digests into.
+  template <class T, class Ts>
+  void hash(const T& x, Ts& xs) {
+    auto d1 = detail::seeded_hash<HashFunction>{seed1_}(x);
+    auto d2 = detail::seeded_hash<HashFunction>{seed2_}(x);
+    for (size_t i = 0; i < xs.size(); ++i)
+      xs[i] = d1 + i * d2;
+  }
+
+  // -- concepts -------------------------------------------------------------
+
+  friend bool operator==(const double_hasher& x, const double_hasher& y) {
+    return x.size() == y.size() && x.seed1_ == y.seed1_ && x.seed2_ == y.seed2_;
+  }
+
+  template <class Inspector>
+  friend auto inspect(Inspector& f, double_hasher& x) {
+    return f(static_cast<super&>(x), x.seed1_, x.seed2_);
```

Comment:
```suggestion
    return f(caf::meta::type_name("double_hasher"), static_cast<super&>(x), x.seed1_, x.seed2_);
```

Reply:
This will likely need clang-format to be applied, the line is too long with my suggested change.